packages/common-library/src/common_library/pydantic_validators.py

```diff
@@ -1,10 +1,35 @@
 import datetime
+import re
 import warnings
 from datetime import timedelta
 
 from pydantic import TypeAdapter, field_validator
 
 
+def _validate_legacy_timedelta_str(time_str: str | timedelta) -> str | timedelta:
+    if not isinstance(time_str, str):
+        return time_str
+
+    # Match the format [-][DD ][HH:MM]SS[.ffffff]
+    match = re.match(
+        r"^(?P<sign>-)?(?:(?P<days>\d+)\s)?(?:(?P<hours>\d+):)?(?:(?P<minutes>\d+):)?(?P<seconds>\d+)(?P<fraction>\.\d+)?$",
+        time_str,
+    )
+    if not match:
```

Comment:
why is this one needed?
pydantic handles this format for date times: YYYY-MM-DD[T]HH:MM[:SS[.ffffff]][Z or [±]HH[:]MM], [-][DD ][HH:MM]SS[.ffffff]
[-][DD ][HH:MM]SS[.ffffff], [±]P[DD]DT[HH]H[MM]M[SS]S (ISO 8601 format for timedelta)

Reply:
This is not supported. That's why I would also remove all these "legacy" solutions and migrate all the _intervals_ to appropriate standard format.

https://github.com/giancarloromeo/osparc-simcore/blob/ded3b252c6012574739ddc30f480147e1f3468f0/services/web/server/tests/unit/isolated/test_studies_dispatcher_settings.py#L33